src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/StartupProjectHelper.cs

```diff
@@ -33,29 +39,24 @@ public StartupProjectHelper([Import(typeof(SVsServiceProvider))] IServiceProvide
         /// </summary>
         public ImmutableArray<T> GetExportFromDotNetStartupProjects<T>(string capabilityMatch) where T : class
         {
-#pragma warning disable RS0030 // Do not used banned APIs
```

Comment:
I know it didn't before, but should this switch to the main thread, just in case? Thats kinda the rule when using IVsUIService, no?

Reply:
There's two rules;

- Either you are async and switch to the UI thread
- Or you are synchronous and throw if you are not the UI thread

This now does the latter by its usage of IVsUIService<T>. If I was writing this - I would have an explicit verification that I'm the UI thread, but I was just removing IServiceProvider usage.